src/interceptors/add_grant_type_to_params.js

```diff
@@ -0,0 +1,23 @@
+/**
+   See what credentials (`username`, `password`, and `authorizationCode`) are
+   passed in params and set the grant_type based on those.
+
+   Changes to `ctx`:
+
+   - add `params.grant_type`
+ */
+export default class AddGrantTypeToParams {
+  enter({ params, ...ctx }) {
+    const { username, password, code } = params;
+
+    if (username && password) {
+      return { ...ctx, params: { grant_type: 'password', ...params } };
+    }
+
+    if (code) {
+      return { ...ctx, params: { grant_type: 'authorization_code', ...params } };
```

Comment:
Is it possible that params (now or in the future contains `grant_type`? Because if it does, this order of spread operator is overwriting going to overwrite hard-coded grant_type.

```js
> const params = { a: 1, b: 2, c: 3 };
> { a: 0, ...params }
// => {a: 1, b: 2, c: 3}
> { ...params, a: 0 }
// => {a: 0, b: 2, c: 3}
```

Reply:
There are quite many places where you have used this overwrite order - so, if you decide to change this, check the other places too.

Reply:
Good point. I'm not actually sure if it should be possible to override `scope` and `grant_type` params in the method invocation. I could actually leave this as it is. Passing those attributes in params to the method is not/will not be documented, so there is no support for them but I'm not sure if some clients could benefit from using them.

Reply:
@Gnito with the current SDK version the `grant_type` and `scope` params can be overwritten by passing them to the `login` method so retaining that behaviour.